search/Bfs.cc

```diff
@@ -171,13 +172,17 @@ BfsIterator::visitParallel(Level to_level,
 {
   size_t thread_count = thread_count_;
   int visit_count = 0;
+  bool is_path_visitor = dynamic_cast<PathVisitor *>(visitor) != nullptr;
   if (!empty()) {
     if (thread_count == 1)
       visit_count = visit(to_level, visitor);
     else {
       std::vector<VertexVisitor*> visitors;
-      for (int k = 0; k < thread_count_; k++)
-	visitors.push_back(visitor->copy());
+      for (int k = 0; k < thread_count_; k++) {
```

Comment:
The caches should not require any changes to Bfs.cc.
That is what class member functions are for.
This is a pretty flagrant violation of the api. Any time you start doing dynamic_cast/if object something is probably wrong.